tests/integration/suite/actors/call/non200.go

```diff
@@ -0,0 +1,109 @@
+/*
+Copyright 2025 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implieh.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package call
+
+import (
+	"context"
+	"fmt"
+	"io"
+	nethttp "net/http"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"google.golang.org/grpc/codes"
+	"google.golang.org/grpc/status"
+
+	rtv1 "github.com/dapr/dapr/pkg/proto/runtime/v1"
+	"github.com/dapr/dapr/tests/integration/framework"
+	"github.com/dapr/dapr/tests/integration/framework/client"
+	"github.com/dapr/dapr/tests/integration/framework/process/daprd/actors"
+	"github.com/dapr/dapr/tests/integration/suite"
+)
+
+func init() {
+	suite.Register(new(non200))
+}
+
+type non200 struct {
+	actors1 *actors.Actors
+	actors2 *actors.Actors
+	called  atomic.Int64
+}
+
+func (n *non200) Setup(t *testing.T) []framework.Option {
+	n.actors1 = actors.New(t,
+		actors.WithActorTypes("abc"),
+		actors.WithActorTypeHandler("abc", func(w nethttp.ResponseWriter, _ *nethttp.Request) {
+			n.called.Add(1)
+			w.WriteHeader(nethttp.StatusInternalServerError)
+			w.Write([]byte("custom error"))
+		}),
+	)
+	n.actors2 = actors.New(t,
+		actors.WithPeerActor(n.actors1),
+	)
+
+	return []framework.Option{
+		framework.WithProcesses(n.actors1, n.actors2),
+	}
+}
+
+func (n *non200) Run(t *testing.T, ctx context.Context) {
+	n.actors1.WaitUntilRunning(t, ctx)
+	n.actors2.WaitUntilRunning(t, ctx)
+
+	for i, actor := range []*actors.Actors{n.actors1, n.actors2} {
+		_, err := actor.GRPCClient(t, ctx).InvokeActor(ctx, &rtv1.InvokeActorRequest{
+			ActorType: "abc",
+			ActorId:   "xyz",
+			Method:    "foo",
+		})
+		require.Error(t, err)
+		status, ok := status.FromError(err)
+		require.True(t, ok)
+		assert.Equal(t, "error invoke actor method: error from actor service: (500) custom error", status.Message())
+		assert.Equal(t, codes.Internal, status.Code())
+		assert.Equal(t, int64(i+1), n.called.Load())
+	}
+
+	time.Sleep(time.Second * 3)
```

Comment:
Rather than putting a sleep here (and below), can we wrap this in an assert eventually?

Reply:
No because we are asserting that it doesn't change

Reply:
ah ok